src/api.js

```diff
@@ -0,0 +1,51 @@
+import axios from 'axios'
+
+const dayms = 24 * 60 * 60 * 1000 // good enough for javascript
+
+// set a cookie with a default expiry of a week
+function setCookie (name, value, days = 7) {
+  const maxAge = days ? `; max-age=${days * dayms}` : ''
+  document.cookie = `${name}=${encodeURIComponent(value)}${maxAge}; path=/`
+}
+
+// get a cookie aspect by name
+function getCookie (name) {
+  for (const section of document.cookie.split(';')) {
+    const [rawName, rawValue] = section.split('=')
+    const thisName = rawName.trim()
+    if (name === thisName) return decodeURIComponent(rawValue)
+  }
+}
+
+// returns the current JWT or null
+function getCurrentJWT () {
```

Comment:
I don't think this is necessary? By spec it's passed to the client as an `httponly,secure` cookie so I'm pretty sure all you can do from the clientside JS is determine if it exists at all.

Reply:
We should probably also add CSRF protection sooner rather than later but that can wait a tiny bit.

Reply:
From the spec:

> In response, the client will receive a response with a [JWT Payload] that may be stored in a browser secret cookie (but should not be stored on the server).

```
HTTP/2 200 OK

{
    "roles": ["scope1", "scope2", {"scope3": "subrole"}] as defined in [scopes](#scopes)"
}
```

That doesn't indicate how it's passed to the client. A JWT is just data; what we do with the data depends on the architecture, which the API spec only partially elucidates. If I could see the backend code, that would really help disambiguate stuff like this.